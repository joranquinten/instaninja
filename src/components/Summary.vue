<template>
  <v-sheet class="summary" elevation="10" rounded v-if="dimensions">
    <div v-html="`<p>${original} ${result}</p><p>${nextStep}</p>`"></div>
  </v-sheet>
</template>
<script>
export default {
  props: {
    dimensions: {
      type: Object,
      required: true,
    },
  },
  computed: {
    original() {
      const { original, aspectRatio } = this.dimensions;
      const addendum =
        aspectRatio.original >= 2
          ? ` It's too wide for a square. Sooooo let's slice it up! ✂️`
          : ``;
      return `I've detected a pretty ${original.mode} photo with a resolution of <strong>${original.width}px</strong> ✕ <strong>${original.height}px</strong> 👀.${addendum}`;
    },
    result() {
      const {
        crop: { squares },
      } = this.dimensions;
      return squares.length <= 1
        ? "I will <strong>not</strong> slice into multiple squares, but I'll do you the favor of creating a square out of the original with a blurred background. 👌"
        : `I will slice this into <strong>${squares.length}</strong> squares and will create a square out of the original with a blurred background. 🤙`;
    },
    nextStep() {
      return `Pick a new photo, or click the Generate button to unleash slashing ninjas on your precious work of art.`;
    },
  },
};
</script>

<style>
.summary {
  margin: 2em 0;
  padding: 1em;
}
</style>
