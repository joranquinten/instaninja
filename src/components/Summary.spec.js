import { shallowMount } from "@vue/test-utils";

import Summary from "./Summary";

import IMAGES from "../../test/stubs/image";

describe("components/Summary", () => {
  Object.keys(IMAGES).forEach(type => {
    test(`should render a summary in "${type}" mode`, () => {
      const wrapper = shallowMount(Summary, { propsData: IMAGES[type] });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
