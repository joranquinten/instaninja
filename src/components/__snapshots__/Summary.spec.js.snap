// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Summary should render a summary in "landscape" mode 1`] = `
<v-sheet
  class="summary"
  rounded=""
>
  <div>
    <p>
      I've detected a pretty landscape photo with a resolution of 
      <strong>
        300px
      </strong>
       ✕ 
      <strong>
        200px
      </strong>
       👀. I will 
      <strong>
        not
      </strong>
       slice into multiple squares, but I'll do you the favor of creating a square out of the original with a blurred background. 👌 
    </p>
    <p>
      Pick a new photo, or click the Generate button to unleash slashing ninjas on your precious work of art.
    </p>
  </div>
</v-sheet>
`;

exports[`components/Summary should render a summary in "portrait" mode 1`] = `
<v-sheet
  class="summary"
  rounded=""
>
  <div>
    <p>
      I've detected a pretty portrait photo with a resolution of 
      <strong>
        200px
      </strong>
       ✕ 
      <strong>
        300px
      </strong>
       👀. I will 
      <strong>
        not
      </strong>
       slice into multiple squares, but I'll do you the favor of creating a square out of the original with a blurred background. 👌 
    </p>
    <p>
      Pick a new photo, or click the Generate button to unleash slashing ninjas on your precious work of art.
    </p>
  </div>
</v-sheet>
`;

exports[`components/Summary should render a summary in "square" mode 1`] = `
<v-sheet
  class="summary"
  rounded=""
>
  <div>
    <p>
       Did you realise that it's already perfectly square? Not going to slice anything! 
    </p>
    <p>
      Please pick a new photo
    </p>
  </div>
</v-sheet>
`;

exports[`components/Summary should render a summary in "wide" mode 1`] = `
<v-sheet
  class="summary"
  rounded=""
>
  <div>
    <p>
      I've detected a pretty landscape photo with a resolution of 
      <strong>
        1200px
      </strong>
       ✕ 
      <strong>
        200px
      </strong>
       👀. It's too wide for a square. Sooooo let's slice it up! ✂️ I will slice this into 
      <strong>
        6
      </strong>
       squares and will create a square out of the original with a blurred background. 🤙 
    </p>
    <p>
      Pick a new photo, or click the Generate button to unleash slashing ninjas on your precious work of art.
    </p>
  </div>
</v-sheet>
`;
